### PR TITLE
Add logic for determining which relations are ambiguous to schema spec

### PR DESCRIPTION
**Pull request by tomhoule**
This engine PR https://github.com/prisma/prisma-engine/pull/82 introduces two changes:

- Fix false positives for the validation on ambiguous relations (this was a bug).
- Make the validation for ambiguous self-relations stricter.  *Some schemas that were valid will be rejected by validation after this change*.

Example schema that is no longer valid:

```
model Human {
  id Int @id
  parent Human
  child Human
}
```

We no longer assume that `parent` and `child` are both ends of the same relation. If that is the desired behaviour, the user now needs to give a name to the relation (e.g. `parent Human @relation("ParentToChild")`).

This PR adds the logic for deciding which relations need to be disambiguated to the schema spec

### CODE REVIEW

File: schema/Readme.md [hunk 1]
```diff
@@ -688,12 +690,22 @@ Specifies a default value if null is provided
...
 - references: _(optional)_ list of field names to reference
 - onDelete: _(optional)_ defines what we do when the referenced relation is deleted
   - **CASCADE**: also delete this entry
   - **SET_NULL**: set the field to null. This is the default
 
+###### Validation
+
+- Ambiguous relations: when one model contains two fields with an `@relation` directive pointing to another model, and both fields have the same relation name, or no relation name, the relation cannot be resolved and a validation error is emitted.
+- Ambiguous self relations: when one model contains two fields referencing the model itself without relation name to disambiguate that they should be seen as the same relation, they are considered ambiguous.
+- Named self relations with more than two fields are rejected, because there is no way to interpret them that makes sense.
```
Comment (tomhoule) [inline schema/Readme.md]: It applies to all named relations so we could remove "self" here, right?

File: schema/Readme.md [hunk 2]
```diff
@@ -688,12 +690,22 @@ Specifies a default value if null is provided
 
 Disambiguates relationships when needed
 
-- name: _(optional)_ defines the name of the relationship
+###### Arguments
+
+- name: _(optional, except when required for disambiguation)_ defines the name of the relationship. The name needs to be disambiguated in the following cases:
```
Comment (mavilein) [inline schema/Readme.md]: It's not the name that is disambiguated. The relations get disambiguated and the name is the tool to do so.

File: schema/Readme.md [hunk 3]
```diff
@@ -688,12 +690,22 @@ Specifies a default value if null is provided
...
+  - There is more than one relation between two types.
+  - The relation is a self-relation with two fields.
```
Comment (mavilein) [inline schema/Readme.md]: I think we can condense this to one rule: A relation is considered ambiguous if a model has 2 fields that refer to the same model. This covers normal and self relations imo.

File: schema/Readme.md (hunk 2 above)
Comment (mavilein) [reply schema/Readme.md]: I think general sentence about ambiguity would be great. Somethink that explains that we need to know which 2 fields belong to the same relation.

**Comment by tomhoule** [comment]
I was thinking we could add an example of ambiguous relationship. Would that be too verbose?

File: schema/Readme.md (hunk 1 above)
Comment (mavilein) [reply schema/Readme.md]: yes

**Pull request merged**